aes/aes-soft/src/fixslice.rs

```diff
@@ -851,89 +864,142 @@ fn double_shiftrows(state: &mut State) {
+        rkeys[off_i] |= (rkeys[off_i - idx_xor] ^ rkeys[off_i] << 2) & 0x30303030;
+        rkeys[off_i] |= (rkeys[off_i - idx_xor] ^ rkeys[off_i] << 2) & 0xc0c0c0c0;
     }
 }
 
-/// Packs two 128-bit input blocs in0, in1 into the 256-bit internal state out
-/// where the bits are packed as follows:
-///
-/// ```text
-/// out[0] = b_24 b_56 b_88 b_120 || ... || b_0 b_32 b_64 b_96
-/// out[1] = b_25 b_57 b_89 b_121 || ... || b_1 b_33 b_65 b_97
-/// out[2] = b_26 b_58 b_90 b_122 || ... || b_2 b_34 b_66 b_98
-/// out[3] = b_27 b_59 b_91 b_123 || ... || b_3 b_35 b_67 b_99
-/// out[4] = b_28 b_60 b_92 b_124 || ... || b_4 b_36 b_68 b_100
-/// out[5] = b_29 b_61 b_93 b_125 || ... || b_5 b_37 b_69 b_101
-/// out[6] = b_30 b_62 b_94 b_126 || ... || b_6 b_38 b_70 b_102
-/// out[7] = b_31 b_63 b_95 b_127 || ... || b_7 b_39 b_71 b_103
-/// ```
+/// Packs two 128-bit input blocks input0, input1 into the 256-bit internal state.
 fn packing(output: &mut [u32], input0: &[u8], input1: &[u8]) {
     debug_assert_eq!(output.len(), 8);
     debug_assert_eq!(input0.len(), 16);
     debug_assert_eq!(input1.len(), 16);
 
-    for (n, (i0, i1)) in input0.chunks(4).zip(input1.chunks(4)).enumerate() {
-        output[n * 2] = u32::from_le_bytes(i0.try_into().unwrap());
-        output[(n * 2) + 1] = u32::from_le_bytes(i1.try_into().unwrap());
-    }
-
-    swapmove!(output[1], output[0], 0x55555555, 1);
-    swapmove!(output[3], output[2], 0x55555555, 1);
-    swapmove!(output[5], output[4], 0x55555555, 1);
-    swapmove!(output[7], output[6], 0x55555555, 1);
-    swapmove!(output[2], output[0], 0x33333333, 2);
-    swapmove!(output[3], output[1], 0x33333333, 2);
-    swapmove!(output[6], output[4], 0x33333333, 2);
-    swapmove!(output[7], output[5], 0x33333333, 2);
-    swapmove!(output[4], output[0], 0x0f0f0f0f, 4);
-    swapmove!(output[5], output[1], 0x0f0f0f0f, 4);
-    swapmove!(output[6], output[2], 0x0f0f0f0f, 4);
-    swapmove!(output[7], output[3], 0x0f0f0f0f, 4);
+    // Bitslicing is a bit index manipulation. 256 bits of data means each bit is positioned at an
+    // 8-bit index. AES data is 2 blocks, each one a 4x4 column-major matrix of bytes, so the
+    // index is initially ([b]lock, [c]olumn, [r]ow, [p]osition):
+    //     b0 c1 c0 r1 r0 p2 p1 p0
+    //
+    // The desired bitsliced data groups first by bit position, then row, column, block:
+    //     p2 p1 p0 r1 r0 c1 c0 b0
+
+    // Interleave the columns on input (note the order of input)
+    //     b0 c1 c0 __ __ __ __ __ => c1 c0 b0 __ __ __ __ __
+    let mut t0 = LE::read_u32(&input0[0x00..0x04]);
+    let mut t2 = LE::read_u32(&input0[0x04..0x08]);
+    let mut t4 = LE::read_u32(&input0[0x08..0x0c]);
+    let mut t6 = LE::read_u32(&input0[0x0c..0x10]);
+    let mut t1 = LE::read_u32(&input1[0x00..0x04]);
+    let mut t3 = LE::read_u32(&input1[0x04..0x08]);
+    let mut t5 = LE::read_u32(&input1[0x08..0x0c]);
+    let mut t7 = LE::read_u32(&input1[0x0c..0x10]);
```

Comment:
I'd prefer to use the `core` versions of these methods with the eventual goal of eliminating the `byteorder` crate as a dependency.

Reply:
Could you spell out the exact method please - not familiar with the libraries at all!

Reply:
`u32::from_le_bytes`, as previously used here:

https://github.com/RustCrypto/block-ciphers/pull/179/files#diff-47f829caba3349d93fab6c8d368ab0fd8b20f3f0a6d95c0a4b8fe51ba7614fa1L891-L892

Reply:
Should be fixed now.